pandas/tests/indexes/categorical/test_indexing.py

```diff
@@ -290,6 +290,18 @@ def test_where(self, klass):
         result = i.where(klass(cond))
         tm.assert_index_equal(result, expected)
 
+    def test_where_non_categories(self):
+        ci = CategoricalIndex(["a", "b", "c", "d"])
+        mask = np.array([True, False, True, False])
+
+        msg = "Cannot setitem on a Categorical with a new category"
```

Comment:
this is an odd message for this operation.

Reply:
yah thats a side-effect of de-duplicating all the validators.  the datetimelike one is nicer, so im planning on doing a pass to make the categorical one behave more like the datetimelike one.  that'll be a big diff since it will involve changing a lot of tests, so planning to do that in a dedicated PR